.github/workflows/tests.yaml

```diff
@@ -82,21 +82,27 @@ jobs:
           ~/.composer/vendor/bin/php-coveralls -v
 
   functional-tests:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     env:
       SELENIUM_SERVER_DOWNLOAD_URL: http://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
 
     strategy:
       fail-fast: false
       matrix:
+        os: ['ubuntu-latest']
         browser: ['chrome', 'firefox']
+        experimental: [false]
         selenium-server: [true, false] # Whether to run via Selenium server or directly via browser driver
         w3c: [true] # Although all builds negotiate protocol by default, it implies W3C protocol for both Chromedriver and Geckodriver
         include:
           # Force OSS (JsonWire) protocol on ChromeDriver - to make sure we keep compatibility
-          - { browser: 'chrome', selenium-server: false, w3c: false }
+          - { browser: 'chrome', os: 'ubuntu-latest', selenium-server: false, w3c: false, experimental: false }
+          - { browser: 'safari', os: 'macos-11.0', selenium-server: false, w3c: false, experimental: true }
+          - { browser: 'safari', os: 'macos-latest', selenium-server: false, w3c: true, experimental: true }
+
+    continue-on-error: ${{ matrix.experimental }}
```

Comment:
I think we should try to make the build passing before it is merged, and not add this.

Reply:
My initial idea was to add macOS as experimental (allow to fail) and try to fix tests in next PR.
To avoid spending too much time on this PR and allow others to contribute.

Reply:
And here is one example:
https://feedbackassistant.apple.com/feedback/9051272

Safaridriver does not persist "pressed" state of Shift

Reply:
The URL requires Apple ID to login...

When there are some tests which do not pass on some browser, we mark them explicitly with `@group exclude-[browser]`, in this case it will be `@group exclude-safari`.

 We have such cases for other browsers as well (chrome, firefox, edge), [for example here](https://github.com/php-webdriver/php-webdriver/blob/a82056dfe7bb1dc3c1d912501f55cd06da5e3cf5/tests/functional/RemoteKeyboardTest.php#L13-L17). Ideally, there should be also a link to some safaridriver bug tracking, if possible.

Also `exclude-safari` should be added [below](https://github.com/php-webdriver/php-webdriver/blob/main/.github/workflows/tests.yaml#L158-L159) to the GH Actions configuration.

When we mark all problematic tests by the group, and thanks to this, it will cause tests not working on safari to be skipped (so they won't fail). So it will allow as to not use the `continue-on-error` at all, am I right? Because the safari tests will be working as the others do.